src/mechanics/ingame/demonslayer/giyu/TomiokaGiyu.java

```diff
@@ -1,37 +1,82 @@
 package mechanics.ingame.demonslayer.giyu;
 
+import mechanics.Texts;
 import mechanics.abilities.Ability;
 import mechanics.characters.Character;
 
 public class TomiokaGiyu extends Character {
+	
+	private int instantKillBoostFromAbility1;
+	private int instantKillBoostFromAbility2;
 
 	public TomiokaGiyu(int id) {
 		super(id);
-		// TODO Auto-generated constructor stub
+
+		this.setA1(new DeadCalm(this.getAbilities()[0]));
+		this.setA2(new SurfaceSlash(this.getAbilities()[1]));
+		this.setA3(new FinalBeheading(this.getAbilities()[2]));
+		this.setA4(new DefensiveStance(this.getAbilities()[3]));
+		
+		this.instantKillBoostFromAbility1 = 0;
+		this.instantKillBoostFromAbility2 = 0;
 	}
 
 	@Override
-	public void applyAbility1(Ability a, Character c) {
-		// TODO Auto-generated method stub
+	public void applyAbility1(Ability ability, Character target) {
+		if (this.instantKillBoostFromAbility1 == 0) {
+			String[] abilityText = {"Water Breathing 11th Form: Dead Calm is improved"};
+			String[] abilityDuration = {Texts.INFINITE};
+
+			target.addActiveSkill(this, ability, abilityText, abilityDuration, false);	
```

Comment:
Here, the target in the argument is the enemy Character. 

The active skill should be added on our Character (Giyu), like this:
```suggestion
			this.addActiveSkill(this, ability, abilityText, abilityDuration, false);	
```